core/src/test/scala/wartremover/warts/StringPlusAnyTest.scala

```diff
@@ -38,9 +38,16 @@ class StringPlusAnyTest extends FunSuite with ResultAssertions {
     assertEmpty(result)
   }
 
+  test("inserting into a Set is allowed") {
+    val result = WartTestTraverser(StringPlusAny) {
+      Set("") + ""
+    }
+    assertEmpty(result)
+  }
+
   test("custom-defined + is allowed") {
     val result = WartTestTraverser(StringPlusAny) {
-      class C { def +(s: String) = s }
```

Comment:
If user creates a method like `def +(s: String): String = ...`, it feels wrong. I would like to trigger a warning in this case anyway. And if user persists in shooting his leg, he might as well disable the rule. But that's my personal opinion.